### DISCUSSION TITLE
I  want to use raylib to build a infinit large 2D scene，how should i do？

### DISCUSSION THREAD

**Discussion by aliule2020** [raysan5/raylib#2207] [General]
I  want to use raylib to build a infinit large 2D scene，how should i do？

**Comment by aliule2020** [comment]
I want to draw shape everywhere，I have tried to use BeginMode2D(Camera2D camera) and EndMode2D(void) with adjusting the camera, but it seems do not work.

**Comment by chriscamacho** [comment]
You'll want to "clip" what you draw to include only the area the camera can see.

You'll need some procedurally generated map, probably with each "cell" being generated using the X and Y coordinate values (concatenated integers, not simple addition so X+Y != Y+X) - usually as a seed to a repeatable random number generator...

I have no idea what "seems do not work" so maybe meet us half way with a better description of what you've tried and what errors or results you encountered....

**Reply by aliule2020** [reply]
Thank you very much, but  I have use function GetScreenToWorld2D solve the problem
